src/parsec.rs

```diff
@@ -63,11 +99,19 @@ impl ParsecMap {
         pub_id: id::PublicId,
         log_ident: &LogIdent,
     ) -> (Option<DirectMessage>, bool) {
-        let parsec = if let Some(parsec) = self.map.get_mut(&msg_version) {
-            parsec
-        } else {
-            return (None, false);
-        };
+        let (parsec, size_counter) =
+            if let Some((parsec, size_counter)) = self.map.get_mut(&msg_version) {
```

Comment:
better have a helper function to de-duplicate the this block (102-114) of code and the below.